src/groups/group.jl

```diff
@@ -274,32 +284,37 @@ function isapprox(
     X,
     Y;
     kwargs...,
-) where {𝔽,O}
+) where {𝔽,O<:AbstractGroupOperation}
     return isapprox(G, identity_element(G), X, Y; kwargs...)
 end
+function Base.isapprox(G::AbstractGroupManifold, p, e::Identity; kwargs...)
+    return isapprox(G, e, p; kwargs...)
+end
+Base.isapprox(::AbstractGroupManifold, ::Identity, ::Identity; kwargs...) = false
 
-Base.show(io::IO, ::Identity{O}) where {O} = print(io, "Identity($O)")
+function Base.show(io::IO, ::Identity{O}) where {O<:AbstractGroupOperation}
+    return print(io, "Identity($O)")
+end
 
-function check_point(G::AbstractGroupManifold{𝔽,O}, e::Identity{O}; kwargs...) where {𝔽,M,O}
+function check_point(
+    G::AbstractGroupManifold{𝔽,O},
+    e::Identity{O};
+    kwargs...,
+) where {𝔽,M,O<:AbstractGroupOperation}
     return nothing
 end
 
 function check_point(
     G::AbstractGroupManifold{𝔽,O1},
     e::Identity{O2};
     kwargs...,
-) where {𝔽,M,O1,O2}
```

Comment:
I am not sure here, is it good to enforce the subtypes or can one be lazy and at the same time allow for Identity to maybe be also used more general?

Reply:
I've noticed that inconsistent usage of subtypes here can cause ambiguities, that is adding `<:AbstractOperation` resolved some ambiguities.

Reply:
Well it should capture different subtypes O1, O2, and there is no possibility to do something more general than `<: AbstractGroupOperation` here, so that was why I was surprised, that that might cause ambiguities.

Reply:
I was also surprised, but see this small example:
```julia
julia> struct A{T<:Number} end

julia> f(::Float64, ::A{T}) where {T} = 11
f (generic function with 1 method)

julia> f(::Number, ::A{T}) where {T<:Number} = 10
f (generic function with 2 methods)

julia> f(0.1, A{Int}())
ERROR: MethodError: f(::Float64, ::A{Int64}) is ambiguous. Candidates:
  f(::Float64, ::A{T}) where T in Main at REPL[2]:1
  f(::Number, ::A{T}) where T<:Number in Main at REPL[3]:1
Possible fix, define
  f(::Float64, ::A{T}) where T<:Number
Stacktrace:
 [1] top-level scope
   @ REPL[4]:1
```

Reply:
Ah, ok, interesting.